### PR TITLE
zsh tab completion script

### PR DESCRIPTION
**Pull request by mestia**
This PR adds zsh tab-completion script.
Since it is outside of main rex code, it shouldn't break anything.

### CODE REVIEW

**Comment by mestia** [comment]
> Thanks, @mestia! Great timing as well, since I have some todos coming up regarding the bash completion script anyway. Unfortunately this PR wasn't discussed in an issue beforehand, so right now I can only give a quick early feedback based on my first thoughts and impressions.

Ok, sorry, next time I'll start with an issue, shall I move the discussion about this PR to the issues now?

> 
> Please note that I'm not using zsh, so some of my points might make no sense :)
> 
> Thinking longer term, I also consider splitting the completion scripts off from the main repo into their own projects, or even better, contributing them to the more offical completion repos for [bash](https://github.com/scop/bash-completion) and [zsh](https://github.com/zsh-users/zsh-completions). That would ensure long-term maintenance better than here in this repo. Perhaps you already considered contributing it there as well?

I see, no, I didn't consider adding them to the official repos, at least not any time soon due to ENOTIME :)

> 
> Either way, thanks for sharing, so people can find it, and start using it +1

I've updated both bash and zsh completions to provide the same functionality.

File: misc/rex.zsh-completion [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+_hostgroups(){
+	grpsfile='hosts.ini'
+	if [[ -e $grpsfile ]]; then
+		hostgroups=(${(f)"$(perl -ne 's/^\[(.*)\].*/$1/g && print' "$grpsfile")"})
+		_wanted hostgroups expl "available host group" compadd -a hostgroups
+	fi
+}
+
+local arguments
+arguments=(
```
[the remark this answers is not in the document]
Comment (mestia) [reply misc/rex.zsh-completion:30]: Exactly, options need to be changed manually, but since most of the job has been done, it should be easy.

**Comment by ferki** [comment]
> Ok, sorry, next time I'll start with an issue, shall I move the discussion about this PR to the issues now?

Nah, I believe no need for that now. Creating an issue first helps the workflow, but I don't see any point to forcibly do all steps backwards - let's just consider the first step skipped :)

> > Perhaps you already considered contributing it there as well?

> I see, no, I didn't consider adding them to the official repos, at least not any time soon due to ENOTIME :)

Alright, let's handle the completion scripts as part of this repo for now, and then we'll see about donating them to other projects later.

> I've updated both bash and zsh completions to provide the same functionality.

Great, thanks!

Though I have to admit I find the bash-completion changes confusing in a Zsh-specific PR (I guess this is what the issue-first approach might help to avoid). They seem to be two logical changes on a single commit, removing the shebang (?), plus adding an unnecessary empty line. It feels like it's best to review/discuss them separately. Could you remove the Bash completion changes and send them on another PR, please?

Testing: as I don't use/have Zsh, I'll let people on IRC know about this PR, to get further testing and feedback.

Squashing: the Zsh script is brand new, so all the Zsh-specific commits could go into a single "Add initial ZSH completion script" commit. Please squash or "fixup" them, and use that as a commit message (I would do it anyway during merge time).

**Comment by mestia** [comment]
Ok,I'll close this PR for now and try to properly address bash and zsh completions in separate PRs.

**Comment by ferki** [comment]
@mestia: if that fits your workflow, it's fine to update this PR for Zsh in-place (by force pushing to the related working branch). So reopen and update could work too.

ps.: GitHub now allows me as maintainer to work directly on PR branches (if not forbidden by the PR author), so if you'd like me to do the squash, I can reopen the PR and update this branch too (assuming this GitHub feature works through close-reopen cycles).

**Comment by mestia** [comment]
I must admit I am quite a noob in github workflows. Let's keep it simple, I've opened 2 new PRs as discussed, squashing the messages and changes into one commit.

**Comment by ferki** [comment]
No worries, thanks for the follow up PRs! I'll try to be more confident to update the PRs in place next time! ^^

**Pull request closed without merging**